tests/unit/test_contrib/test_repository.py

```diff
@@ -29,6 +29,7 @@ def test_app_repository_signature_namespace() -> None:
         "NotInSearchFilter": NotInSearchFilter,
         "FilterTypes": FilterTypes,
     }
+    assert all(app.signature_namespace[k] == v for k, v in expected_namespace.items())
```

Comment:
Any reason for this change? Direct comparison gives nicer outputs in pytest because it then includes a diff.

Reply:
initially, the `OpenAPIPlugin` was adding `OpenAPI` to the signature namespace causing this to fail.

Generally, though, I don't think that tests should assume the signature namespace is empty if they are testing their own interaction with it - because it might not be, especially as we adopt using plugins internally over time.

I can back this out because I've not ended up using DI for the `OpenAPI` object anymore, but even before then I didn't need it because of that whole unconditional type recursion for dataclasses that msgspec does. I was pretty much forced to put either the entire `openapi.spec` namespace into `signature_namespace` to support it wanting to recurse through the subtypes of `OpenAPI`, or type it as `Any` - which is what I did... until I removed it from DI altogether. I think this is a real issue that needs to be addressed.. if a user wants to annotate a handler with a dependency type that they don't control, e.g., comes from a 3rd party, and that 3rd party arbitrarily uses `@dataclass` types and `if TYPE_CHECKING` imports, then it blows up and the user is forced to disable that dependency validation.

Another thing I just thought of is that we should have some internal convention for population of the signature namespace for our internal things, e.g., `from litestar.openapi.spec import OpenAPI as _LS_OpenAPI` or something like that, to reduce the potential for name collisions with user land stuff.

Reply:
> Another thing I just thought of is that we should have some internal convention for population of the signature namespace for our internal things, e.g., `from litestar.openapi.spec import OpenAPI as _LS_OpenAPI` or something like that, to reduce the potential for name collisions with user-land stuff.

Sounds like a good idea!